cf/data/creation.py

```diff
@@ -81,22 +82,46 @@ def to_dask(array, chunks, **from_array_options):
 
     **Examples**
 
-    >>> to_dask([1, 2, 3])
+    >>> cf.data.creation.to_dask([1, 2, 3])
     dask.array<array, shape=(3,), dtype=int64, chunksize=(3,), chunktype=numpy.ndarray>
-    >>> to_dask([1, 2, 3], chunks=2)
+    >>> cf.data.creation.to_dask([1, 2, 3], chunks=2)
     dask.array<array, shape=(3,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>
-    >>> to_dask([1, 2, 3], chunks=2, {'asarray': True})
+    >>> cf.data.creation.to_dask([1, 2, 3], chunks=2, {'asarray': True})
     dask.array<array, shape=(3,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>
+    >>> cf.data.creation.to_dask(cf.dt(2000, 1, 1), 'auto')
+    dask.array<array, shape=(), dtype=object, chunksize=(), chunktype=numpy.ndarray>
+    >>> cf.data.creation.to_dask([cf.dt(2000, 1, 1)], 'auto')
+    dask.array<array, shape=(1,), dtype=object, chunksize=(1,), chunktype=numpy.ndarray>
 
     """
+    if not (
+        is_dask_collection(array)
+        or isinstance(
+            array, (np.ndarray, list, tuple, memoryview) + np.ScalarType
+        )
+    ):
+        if not hasattr(array, "shape"):
+            # 'array' is not of a type that `da.from_array` can cope
+            # with, so convert it to a numpy array
+            array = np.asanyarray(array)
+
     kwargs = from_array_options
     lock = getattr(array, "_dask_lock", False)
     if lock:
         lock = get_lock()
-    #    kwargs.setdefault("lock", getattr(array, "_dask_lock", False))
+
     kwargs.setdefault("lock", lock)
     kwargs.setdefault("meta", getattr(array, "_dask_meta", None))
-    return da.from_array(array, chunks=chunks, **kwargs)
+
+    try:
+        return da.from_array(array, chunks=chunks, **kwargs)
+    except NotImplementedError as e:
+        if (
+            str(e)
+            == "Can not use auto rechunking with object dtype. We are unable to estimate the size in bytes of object data"
```

Comment:
Maybe:

```suggestion
            == "Can not use auto rechunking with object dtype. Unable to estimate the size in bytes of object data"
```

in line with our passive 'house style' and also to be slightly more concise?

Reply:
I agree, but this message is produced by `dask`, and we can't control it. It's not very robust, though, having our logic controlled by the exact format and wording of someone else's error message. How about https://github.com/NCAS-CMS/cf-python/pull/441/commits/6adeaae2d93d7772b809176df2fbcdcc7cd9064f?

Reply:
With the fix: a0a2749b0adf1faac9e982f3ab0951ec60ed8ff4! I realise that autoochunking can still occur if chunks is not `"auto"`. E.g. in ...

```python
>>> da.from_array(np.arange(200).reshape(10, 20), chunks={1: 5})
dask.array<array, shape=(10, 20), dtype=int64, chunksize=(10, 5), chunktype=numpy.ndarray>
```
... dimension `0` was auto-chunked.

Reply:
> I agree, but this message is produced by dask, and we can't control it.

Ah, I see. Not passive enough from Dask then! Far too friendly for my liking :smile: ...

I liked 6adeaae very much; now you suggest it, it seems much more robust to test against the chunking argument. But a0a2749 is certainly sufficient given your final note, so we can consider this resolved.